test/e2e/marketplace_test.go

```diff
@@ -0,0 +1,186 @@
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"testing"
+	"time"
+
+	apis "github.com/operator-framework/operator-marketplace/pkg/apis"
+	operator "github.com/operator-framework/operator-marketplace/pkg/apis/marketplace/v1alpha1"
+
+	framework "github.com/operator-framework/operator-sdk/pkg/test"
+	"github.com/operator-framework/operator-sdk/pkg/test/e2eutil"
+
+	olm "github.com/operator-framework/operator-lifecycle-manager/pkg/api/apis/operators/v1alpha1"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+var (
+	retryInterval        = time.Second * 5
+	timeout              = time.Second * 60
+	cleanupRetryInterval = time.Second * 1
+	cleanupTimeout       = time.Second * 5
+	testNamespace        = "operator-test"
+)
+
+func TestMarketplace(t *testing.T) {
+	operatorsourceList := &operator.OperatorSource{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.OperatorSourceKind,
+			APIVersion: fmt.Sprintf("%s/%s",
+				operator.SchemeGroupVersion.Group, operator.SchemeGroupVersion.Version),
+		},
+	}
+	catalogsourceconfigList := &operator.CatalogSourceConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.CatalogSourceConfigKind,
+			APIVersion: fmt.Sprintf("%s/%s",
+				operator.SchemeGroupVersion.Group, operator.SchemeGroupVersion.Version),
+		},
+	}
+	err := framework.AddToFrameworkScheme(apis.AddToScheme, operatorsourceList)
+	if err != nil {
+		t.Fatalf("failed to add operatorsource custom resource scheme to framework: %v", err)
+	}
+	err = framework.AddToFrameworkScheme(apis.AddToScheme, catalogsourceconfigList)
+	if err != nil {
+		t.Fatalf("failed to add catalogsourceconfig custom resource scheme to framework: %v", err)
+	}
+	// run subtests
+	t.Run("marketplace-group", func(t *testing.T) {
+		t.Run("Cluster", MarketplaceCluster)
+	})
+}
+
+func defaultCreateTest(t *testing.T, f *framework.Framework, ctx *framework.TestCtx) error {
+	namespace, err := ctx.GetNamespace()
+	if err != nil {
+		return fmt.Errorf("could not get namespace: %v", err)
+	}
+
+	testOperatorSource := &operator.OperatorSource{
+		TypeMeta: metav1.TypeMeta{
+			Kind: operator.OperatorSourceKind,
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "global-operators",
+			Namespace: namespace,
+		},
+		Spec: operator.OperatorSourceSpec{
+			Type:              "appregistry",
+			Endpoint:          "https://quay.io/cnr",
+			RegistryNamespace: "operators",
```

Comment:
It might actually make sense for us to have a separate registry specifically for these e2e tests that differs from our example (at least in terms of if and when it is updated).

Reply:
I prefer using a separate namespace for e2e tests and one for development. Not only that, we need to convert that namespace to have individual operator repositories like we are planning to have in the product. Though it wouldn't hurt to have one repository with two or more operators to ensure we don't break that functionality.

Reply:
I believe a free-tier user is allowed to have only one organization, I originally created `operators`. So  either @kevinrizza / @aravindhp creates an org for us or have someone from redhat create an org for us.

Reply:
Please use `marketplace_e2e`.  They have been populated with `federationv2 jboss-middleware mongodb-enterprise service-catalog couchbase-enterprise descheduler dynatrace-monitoring amq-streams etcd prometheus` operators.